CHANGES_NEXT_RELEASE

```diff
@@ -1,2 +1,3 @@
 - Fix: idPattern '.*' is now allowed in NGSIv2 registrations (#3458)
```

Comment:
With regards to documentation I think the following docs should be touched (I'm using RTD references for convenicence, please do the modificaitons in their .md counterparts):

* https://fiware-orion.readthedocs.io/en/master/user/ngsiv2_implementation_notes/index.html#registrations. 
    * From: "Although NGSIv1 is deprecated, for the time being, NGSIv2-based forwarding has not been defined (see this issue about it) so the only valid option is to always use "legacyForwarding": true (otherwise a 501 Not Implemented error response will be the result)."
   * To: "Although NGSIv1 is deprecated, some Context Provider could not has been migrated yet to NGSIv2, so this mode could be useful".
* https://fiware-orion.readthedocs.io/en/master/user/context_providers/index.html. Several actions:
   * Remove "legacyForwardingMode: true" in the example and modify the steps describing the NGSIv1-based forwarding for NGSIv2-based ones.
   * This sentence "First, Orion doesn't include the filter in the POST /v1/queryContext operation forwarded to CP" should use GET /v2/entities instead of POST /v1/queryContext
   * The picture needs to be modified. I can take care of this (I'll pass you the new png file, to be used in the PR)
* https://fiware-orion.readthedocs.io/en/master/admin/database_model/index.html#registrations-collection.
   * From: "format: the format to use to send forwarded requests. The only accepted value for now is JSON (meaning NGSIv1 format), although this may change in the future (see issue about NGSIv2-based forwarding)."
   * To: "format: the format to use to send forwarded requests: JSON (meaning NGSIv1 format, used by legacy CPrs), or "V2""

Reply:
Maybe I have missed some one... A grep -i search in *.md for key words (cpr, registration, provider, forwarding, etc.) could be raise additional places to fix. Please, check it.

Reply:
Docu fixed in 7001535f8.
Found a few places more. Hopefully this is all.
To be completely sure we'd need to read through the entire documentation.
Perhaps we can ask for help from our Japanese co-corker?
I don't mean to read through all of it, but for him, as he started not so long ago, all of this is a lot mor e"fresh" than for us ...

Reply:
More things I have spotted (in the devel manual):

* https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/devel/sourceCode.md#srclibserviceroutines. Remove the "IMPORTANT" paragraph (and join together the line just after it and the line just before it).
* https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/devel/cprs.md (a document in the devels manual devoted to CPr and fowarding!).
    * Remove references to NGSI9 (obsolete term from NGSIv1 days)
    * Changes in the FW-04 description at https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/devel/cprs.md#flow-fw-04. Currently the description only covers NGSIv1 but this method has been expanded to cover also NGSIv2. Hint: try to do it in a way in which changes in the diagram get minimized. If after that modifications in the png are needed I can help with that.
        * Take note also off FW-02. Not for this PR but in the next one for fowared updated it would need also to be modified.

Reply:
(If you need adjustments in the diagramas just tell me so I can provide the new .png files for that)

Reply:
Fixed in 69093c2b8

Reply:
> (If you need adjustments in the diagramas just tell me so I can provide the new .png files for that)

At the end you didn't need a replacing .png? (just to confirm)

Reply:
I have looked this in detail. I think the png should be replaced with this one:

![imagen](https://user-images.githubusercontent.com/1534240/56604262-dd946080-6601-11e9-8bea-386e299dde0a.png)

And FW-04 step 2 description changed to (for instance):

> * Fowarded request has to be built (step 2). In the case of NGSIv1, we need to extract information of the binary object to text to be able to send the REST request (plain text) to the Context Provider using `POST /v1/queryContext`. In the case of NGSIv2, the use `GET /v2/entities` an no payload is required.

Reply:
Fixed in aef7cb8